src/rest_framework_dso/renderers.py

```diff
@@ -82,6 +107,61 @@ class HALJSONRenderer(RendererMixin, renderers.JSONRenderer):
     # Define the paginator per media type.
     compatible_paginator_classes = [pagination.DSOPageNumberPagination]
 
+    def render(self, data, accepted_media_type=None, renderer_context=None):
+        """Render the data as streaming."""
+        if data is None:
+            return
+
+        indent = self.get_indent(accepted_media_type, renderer_context or {})
+        if indent:
+            yield self._render_json_indented(data)
+        else:
+            yield from _chunked_output(self._render_json(data))
+
+    def _render_json_indented(self, data):
+        """Render indented JSON for the browsable API"""
+        if isinstance(data, dict) and (embedded := data.get("_embedded")):
+            # The ReturnGenerator is rendered as empty list, so convert these values first.
+            # Using orjson.dumps(data, default) here doesn't work, as the dict ordering
+            # isn't honored by orjson.
+            for key, value in embedded.copy().items():
+                if inspect.isgenerator(value) or isinstance(value, itertools.chain):
+                    embedded[key] = list(value)
+
+        return orjson.dumps(data, option=orjson.OPT_INDENT_2)
+
+    def _render_json(self, data, level=0):
+        if not data:
+            yield orjson.dumps(data)
+            return
+        elif isinstance(data, dict):
+            # Streaming output per key, which can be a whole list in case of embedding
+            yield b"{"
```

Comment:
Instead of using bytes directly you might want to consider using `.encode()` to convert `str`s to `bytes`. That is arguably more readable.

Reply:
On second thought, `b""` is just fine now that I've found out orjson returns bytes instead of `str` (that's diffferent than what the stdlib json lib does)

Had orjson return str I  would have advocated for doing regular string formatting with a `.encode()` at the end (including for the literals as here). But as orjson returns bytes, using bytes for the literals is better here.